app.js

```diff
@@ -419,6 +419,44 @@ function ApiHandler(compileHandler) {
             }).join("\n"));
         }
     }, this));
+     // ---- this whole thing should be in a separate file etc, proof of concept only
+    // TODO: we'll need something smarter than this; URLs are long-lived and we nede
+    // to think about it; languages...formatters...etc.maybe something like:
+    var exec = require('./lib/exec');
+    this.handler.post('/format/clang-format-3.8', function (req, res) {
```

Comment:
(I can help with the configuration stuff...if you don't get to it I can make the necessary changes)